scripts/compare-memory-usage.ts

```diff
@@ -0,0 +1,18 @@
+const threshold = 90;
+
+const main = () => {
+  const v1MemoryUsage = parseInt(process.argv[2]);
+  const canaryMemoryUsage = parseInt(process.argv[3]);
+  const percentageBasedOnV1 = (canaryMemoryUsage / v1MemoryUsage) * 100;
+  if (percentageBasedOnV1 > threshold) {
+    throw new Error(
+      `Test failed: v1 memory usage is ${v1MemoryUsage} and canary memory usage is ${canaryMemoryUsage} (${percentageBasedOnV1} %). Consider changing the threshold (${threshold} %) or fix it.`
+    );
+  }
+  /* eslint-disable-next-line no-console */
+  console.log(
+    `Test passed: v1 memory usage is ${v1MemoryUsage} and canary memory usage is ${canaryMemoryUsage} (${percentageBasedOnV1} %).`
+  );
+};
+
+main();
```

Comment:
Currently the test-script.js are executed by the memory-test.yml and then they write their result to a file which is then piped into this script. It toook my a few minutes to figure this out and I knew that the memeory workflow is there. I would change it in the following way:
- the `compare-memory-usage.ts` is extended so that it executes the two `test-script.js` files. If you use execa in sequence the heap size should be not affected (not in prallel of course). 
- the `test-script.js` just print and return the used memory.
- the script collects the two values returned by execa execution -> no need to write files to the disk
- evaluation is done.

This would simplify the `memory-tests.yml` and put all the execution steps in one script - please note if that is not possible because the heap tests need a clean isolated execution please ignore this comment.